exercises/bowling/example.py

```diff
@@ -0,0 +1,157 @@
+MAX_PINS = 10
+NUM_FRAMES = 10
+
+
+class BowlingGame(object):
+    def __init__(self):
+        self.rolls = []
+        self.totalScore = 0
+        self.currentFrame = Frame()
+        self.bonusRollsAccrued = 0
+        self.bonusRollsSeen = 0
+
+    def roll(self, pins):
+        if (self.isBonusRoll()):
+            self.bonusRollsSeen += 1
+
+        # is the second roll valid based off the first?
+        if (self.currentFrame.isOpen() and
+           self.currentFrame.getFrame()[0] is not None):
+            if(self.currentFrame.getFrame()[0] + pins > MAX_PINS):
+                raise ValueError
+
+        # open a new frame if the last one has been closed
+        if (self.currentFrame.isOpen() is False):
+            self.currentFrame = Frame()
+
+        # valid roll between 0-10
+        if (pins in range(MAX_PINS + 1)):
+            # raise an error if the game is over and they try to roll again
+            if((len(self.rolls) == NUM_FRAMES) and
+               self.bonusRollsAccrued == 0):
+                raise IndexError
+            else:
+                self.currentFrame.roll(pins,
+                                       self.isBonusRoll(),
+                                       self.bonusRollsAccrued,
+                                       self.bonusRollsSeen)
+                # if we closed it add it to our rolls
+                if (self.currentFrame.isOpen() is False):
+                    self.rolls.append(self.currentFrame)
+                    # If this is the last frame did we earn any bonus rolls?
+                    if (len(self.rolls) == NUM_FRAMES):
+                        self.bonusRollsEarned()
+                        print(self.bonusRollsAccrued)
+        else:
+            raise ValueError
+
+    def score(self):
+        frame_index = 0
+
+        while (frame_index <= NUM_FRAMES-1):
+            print()
+            frame = self.rolls[frame_index].getFrame()
+
+            roll1 = frame[0]
+            roll2 = frame[1]
+
+            if (self.isStrike(roll1)):
+                self.totalScore += roll1 + self.stikeBonus(frame_index)
+
+            else:
+
+                if(self.isSpare(roll1, roll2)):
+                    self.totalScore += roll1 + roll2 + \
+                                       self.spareBonus(frame_index)
+                else:
+                    self.totalScore += roll1 + roll2
+
+            frame_index += 1
+
+        return self.totalScore
+
+    def isStrike(self, pins):
+        if (pins == MAX_PINS):
```

Comment:
It is not necessary to use parenthesis in `if` statements like in `Java`. Add them only to change order precedence or reduce ambiguity